idaes/models/properties/modular_properties/eos/ceos.py

```diff
@@ -447,6 +462,144 @@ def rule_delta_eq(m, p1, p2, p3, i):
+                except KeyError:
+                    rule = MixingRuleA.default
+        
+                a_crit = getattr(m, "a_crit")
+                if rule == MixingRuleA.default:
+                    return rule_am_crit_default(m, cname, a_crit)
+                else:
+                    raise ConfigurationError(
+                        "{} Unrecognized option for Equation of State "
+                        "mixing_rule_a: {}. Must be an instance of MixingRuleA "
+                        "Enum.".format(m.name, rule))
+            b.add_component('am_crit', Expression(rule=rule_am_crit))
+        
+            def rule_bm_crit(m):
+                try:
+                    rule = m.params.get_phase(reference_phase). \
+                        config.equation_of_state_options["mixing_rule_b"]
+                except KeyError:
+                    rule = MixingRuleB.default
+        
+                b = getattr(m, cname+"_b")
+                if rule == MixingRuleB.default:
+                    return sum(m.mole_frac_comp[i] * b[i]
+                                for i in m.component_list)
+                else:
+                    raise ConfigurationError(
+                        "{} Unrecognized option for Equation of State "
+                        "mixing_rule_a: {}. Must be an instance of MixingRuleB "
+                        "Enum.".format(m.name, rule))    
+            b.add_component('bm_crit', Expression(rule=rule_bm_crit))
+    
+            def rule_A_crit(m):
+                am_crit = getattr(m, "am_crit")
+                return EoS_param[ctype]['omegaA'] == \
+                    (am_crit * m.pressure_crit_mix /
+                      (Cubic.gas_constant(b) * m.temperature_crit_mix)**2)
+            b.add_component('A_crit', Constraint(rule=rule_A_crit))
+        
+            def rule_B_crit(m):
+                bm_crit = getattr(m, "bm_crit")
+                return EoS_param[ctype]['coeff_b'] == \
+                    (bm_crit * m.pressure_crit_mix /
+                      (Cubic.gas_constant(b) * m.temperature_crit_mix))
+            b.add_component('B_crit', Constraint(rule=rule_B_crit))
+
+    @staticmethod
+    def initialize_critical_properties(b, p):
+        if p == 'Vap':
+            return None
+        else:
+            if b.params.config.supercritical_extension:
+                b.temperature_crit_mix.value = value(
+                    sum(b.mole_frac_comp[j] *
+                        b.params.get_component(j).temperature_crit
+                        for j in b.component_list))
+        
+                v_crit = value(
+                    sum(b.mole_frac_comp[j] *
+                        b.params.get_component(j).volume_crit
+                        for j in b.component_list))
+                Z_crit = value(
+                    sum(b.mole_frac_comp[j] *
+                        b.params.get_component(j).compress_factor_crit
+                        for j in b.component_list))
```

Comment:
According to the Peng-Robinson EoS, the critical compressibility for every substance is the same: `0.307`. I'm not sure how calculating a different number from material specific data affects your results. Is this coming from a paper?